starter-repos/app-template/config/base/app/deployment.yaml

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: app
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+  template:
+    metadata:
+      name: app
+    spec:
+      containers:
+      - name: app
+        image: gcr.io/jingfangliu-194020/app:v1
```

Comment:
shall we change this to a kpt setter or a env variable?

Reply:
I think we need to change it to a kpt setter after kpt can support kustomize format.

Reply:
I may misunderstand this. This is the deployment resource and kpt setter works on yaml files under a given directory. Why is the kustomize support required.

Reply:
The config is kustomize format, the hydration is done by kustomize build. By kpt setter, I think you assume the hydration is done by kpt.

Reply:
The kpt setter doesn't have dependencies on hydration. 
e.g. add a new setter app-image-url and let users update the app-image-url to their own container registry url. 
```bash
kpt kpt pkg init .
kpt cfg create-setter . app-image "gcr.io/jingfangliu-194020/app:v1"
```

Reply:
> The kpt setter doesn't have dependencies on hydration.

Agree. However, if users manually update the kustomization.yaml file or update it by `kustomize edit`. If they accidentally remove the or change the setter comments, it will break. For the kustomize format, we currently recommend to manually edit the file.

Reply:
For the image tag specifically, we apply a patch in each config/envs/<ENV>. The image tag can be manually changed in the patch. In the future, when we add the kpt setter, it shouldn't be there instead of in the base.

Reply:
kk